src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -450,17 +461,16 @@ static void punishPredictedColumn(
 
 void TemporalMemory::activateCells(const SDR &activeColumns, bool learn) {
     NTA_CHECK( activeColumns.dimensions == columnDimensions_ );
-    const auto &sparse = activeColumns.getSparse();
+    auto &sparse = activeColumns.getSparse();
+    std::sort(sparse.begin(), sparse.end()); //TODO this is constly, do we really need to sort?
     activateCells(sparse.size(), sparse.data(), learn);
 }
 
 void TemporalMemory::activateCells(const size_t activeColumnsSize,
                                    const UInt activeColumns[], bool learn) {
   if (checkInputs_ && activeColumnsSize > 0) {
-    NTA_CHECK(isSortedWithoutDuplicates(activeColumns,
-                                        activeColumns + activeColumnsSize-1))
-        << "The activeColumns must be a sorted list of indices without "
-           "duplicates.";
+    NTA_CHECK(std::is_sorted(activeColumns, activeColumns + activeColumnsSize-1))
+        << "The activeColumns must be a sorted list of indices without duplicates.";
```

Comment:
@ctrl-z-9000-times with regards to #337 , I'm having problems with this code (check), 
calling this from the SDR, the list is never sorted, so I have to cast to vector, sort, and pass to the old-API version. 
Solutions: 
- SDR.getSparse() would return a sorted vector of indices (slower first, but cached) 
- remove this requirement (and check) from activateCells (preferable) 
thoughts?

Reply:
I don't know how much about how sparse data is used in the algorithms but if it is sorted then algorithms can can do merge style processing (single pass) when doing operations on two sparse vectors.
See VectorHelpers::unionOfVectors( )

Reply:
You should be able to sort the data in place. GetSparse returns a non-constant reference.
`std::sort( sdr.getSparse().begin(), sdr.getSparse().end() )`

Reply:
That would work.  if you remove the sorted requirement, you can add that sort to unionOfVectors( ).  I wonder if there is other code that assumes that sparse data is sorted?

Reply:
> remove this [sorted, unique] requirement (and check) from activateCells (preferable)

would not work, `iterGroupBy` later in that function requires this from the data! #263